src/shared/condition.c

```diff
@@ -450,6 +451,81 @@ static int condition_test_architecture(Condition *c, char **env) {
         return a == b;
 }
 
+#define DTCOMPAT_FILE "/sys/firmware/devicetree/base/compatible"
+static int condition_test_firmware_devicetree_compatible(const char *dtcarg) {
+        int err;
```

Comment:
doesn't matter, but throughout the codebase we use "r" as the name for the variable we store function call return values into. Please do the same here, i.e. just rename `err` to `r`